src/Simplex/Chat/Store.hs

```diff
@@ -789,47 +792,66 @@ deleteUserAddress db User {userId} = do
     [":user_id" := userId]
   DB.execute db "DELETE FROM user_contact_links WHERE user_id = ? AND local_display_name = '' AND group_id IS NULL" (Only userId)
 
-getUserAddress :: DB.Connection -> UserId -> ExceptT StoreError IO (ConnReqContact, Bool, Maybe MsgContent)
+data UserContactLink = UserContactLink
+  { -- userContactLinkId :: Int64,
+    connReqContact :: ConnReqContact,
+    autoAccept :: Bool,
+    autoAcceptIncognito :: Bool,
+    autoReply :: Maybe MsgContent
+  }
+  deriving (Show, Generic)
+
+instance ToJSON UserContactLink where toEncoding = J.genericToEncoding J.defaultOptions
+
+getUserAddress :: DB.Connection -> UserId -> ExceptT StoreError IO UserContactLink
 getUserAddress db userId =
   ExceptT . firstRow id SEUserContactLinkNotFound $
-    DB.query
+  map toUserContactLink
+    <$> DB.query
       db
       [sql|
-        SELECT conn_req_contact, auto_accept, auto_reply_msg_content
+        SELECT conn_req_contact, auto_accept, auto_accept_incognito, auto_reply_msg_content
         FROM user_contact_links
         WHERE user_id = ? AND local_display_name = '' AND group_id IS NULL
       |]
       (Only userId)
+    where
+      toUserContactLink :: (ConnReqContact, Bool, Bool, Maybe MsgContent) -> UserContactLink
+      toUserContactLink (connReqContact, autoAccept, autoAcceptIncognito, autoReply) = UserContactLink {connReqContact, autoAccept, autoAcceptIncognito, autoReply}
 
-getUserContactLinkById :: DB.Connection -> UserId -> Int64 -> IO (Maybe (ConnReqContact, Bool, Maybe MsgContent, Maybe GroupId))
+getUserContactLinkById :: DB.Connection -> UserId -> Int64 -> IO (Maybe (UserContactLink, Maybe GroupId))
 getUserContactLinkById db userId userContactLinkId =
   maybeFirstRow id $
-    DB.query
+    map toUserContactLink
```

Comment:
Can't correct code, but instead of

```
maybeFirstRow id $
     map toUserContactLink
       <$>
```

you need 

```
maybeFirstRow toUserContactLink $
```

and that's it